File: 프로그래머스/3/12927. 야근 지수/야근 지수.py
from heapq import heappop, heappush, heapify
def solution(n, works):
    if sum(works) <= n:
        return 0
    
    # max heap
    works = [-x for x in works]
    heapify(works)
    
    for _ in range(n):
        i = heappop(works)
        i +=1
        heappush(works,i)
    
    return sum([x**2 for x in works])

File: 프로그래머스/3/12927. 야근 지수/야근 지수.js
function solution(n, works) {
    if(n >= works.reduce((acc,cur) => acc + cur)) return 0;
    works.sort((a,b) => b-a)
    while(n != 0) {
        const max = works[0]
        for(i=0 ; i<works.length; i++) {
            if(works[i] >= max) {
                n--
                works[i]--
            }
            if(!n) break
        };
    };
    return works.reduce((acc,cur) => acc + cur**2 , 0 )
}